servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1094,4 +1120,30 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
       return mDependencyMap.getPriorityDependencyList();
     }
   }
+
+  /**
+   * MasterInodeTTL periodic check.
+   */
+  public final class MasterInodeTTLCheckExecutor implements HeartbeatExecutor {
+    @Override
+    public void heartbeat()  {
+      synchronized (mInodeTree) {
+        List<Inode> inodes = mInodeTree.getInodeChildrenRecursive(mInodeTree.getRoot());
```

Comment:
One simple way to do this is to have a list of buckets where each bucket has a target TTL range (e.g., [0 , 1 hr) for a bucket, [1 hr, 2hr) for another bucket etc). And periodically update buckets. Note that, usually we treat TTL value as a suggested value, so a file with TTL=100s doesn't need to die after 101s, it just means we keep this file for at least 100 seconds. So we don't have to access and update those buckets too frequently.